phpBB/phpbb/filesystem/filesystem.php

```diff
@@ -742,4 +753,33 @@ protected function resolve_path($path, $prefix = '', $absolute = false, $return_
 	{
 		return helper::resolve_path($path, $prefix, $absolute, $return_array);
 	}
+
+	/**
+	 * Get a temporary directory to write files
+	 *
+	 * @return string	returns the directory
+	 */
+	public function get_temp_dir()
+	{
+		if (!isset($this->temp_dir))
+		{
+			$tmp_dir = (function_exists('sys_get_temp_dir')) ? sys_get_temp_dir() : '';
+
+			// Prevent trying to write to system temp dir in case of open_basedir
+			// restrictions being in effect
+			if (empty($tmp_dir) || !@file_exists($tmp_dir) || !@is_writable($tmp_dir))
+			{
+				$tmp_dir = $this->cache_temp_dir;
+
+				if (!is_dir($tmp_dir))
+				{
+						$this->mkdir($tmp_dir, 0777);
```

Comment:
Why is there a 2 tab indent?

Reply:
Uhm typo.

About this pr, i was thinking about moving this method to another class (dont know how to call it), because i add a parameter to the entire class only for this.